### ISSUE TITLE
Add helper function to anonymize file download logs.

### ISSUE THREAD

**Issue by cklosowski** [easydigitaldownloads/easy-digital-downloads#6500]
## Enhancement Request
File Download logs contain IP addresses, email addresses and customer name. A request to delete the data should anonymize this information, as keeping it in tact partially can help detect fraudulent activity.

### Justification or use case
This is a [GDPR](#6216 ) issue and is related to #6499

**Comment by cklosowski** [comment]
With #6552, our task here is greatly reduced.

If the payment associated was deleted we just need to delete the file download logs.

If the payment is anonymized, we can just anonymize the IP address (since that is the last thing that is identifying in the logs)

If the payment has no action taken, we should take no actions on the logs.

**Comment by SDavisMedia** [comment]
Alright! My test here was similar to the payment anonymizer test, since the logs are based on the payment records. Success across the board! 

I suggest two small changes to the success message after running the eraser:

1. Change `All file download logs anonymized for foo@bar.com` to `All eligible file download logs anonymized or deleted for foo@bar.com`. There are two reasons for this: **for one**, when the associated payment is not being anonymized, neither is the log. So there will be cases when _some_ or _no_ logs are anonymized, [like so](https://cl.ly/3S3k2Y1Q2m1l). We can cover all cases by just referencing eligible logs. **Secondly**, it matches what we already do for `All eligible payments anonymized or deleted for foo@bar.com`. 

2. In a case where all payments are _deleted_ by the eraser, the file download logs are deleted as well. But the success message of the eraser states `No file download logs found for foo@bar.com`, probably because of the order in which things happen. I think that's misleading and could cause users to wonder if there are leftover logs. I think a better message, of _all_ scenarios, would be to stick with my suggestion above: `All eligible file download logs anonymized or deleted for foo@bar.com`.

Basically, regardless of what happened to the payment records (and therefore the file download logs) during the erase, both the payment message and the file download log message should follow the same catch-all format.

`All eligible payments anonymized or deleted for foo@bar.com`
`All eligible file download logs anonymized or deleted for foo@bar.com`

Besides these success message tweaks, all is well for me! :+1:

**Comment by cklosowski** [comment]
@SDavisMedia  The file download logs are deleted from deleted payments as part of the payment process, not the download file log process, so I don't really have any way to actually handle that messaging.

I'll just update all messaging to the to remove the 'None' case, and go to 'All' since hat way it just covers it all.
